### DISCUSSION TITLE
Do I need a TPU if I only have 1 1080p camera?

### DISCUSSION THREAD

**Discussion by anthonyma94** [blakeblackshear/frigate#3739] [Ask A Question] [answered]
As the subject says. TPUs are hard to find in Canada and I was hoping I don't need another expense for this project.

**Comment by NickM-27** [comment, marked as answer]
Mostly depends on your host device (the one running frigate) and your expectations. We have a number of users running 1-2 cameras on a Raspberry Pi but that also includes spikes of CPU usage when object detection is run. 

As the docs say, CPU is only recommended for testing or initial trial runs, you might as well run it and see how you like it and if the performance is acceptable. There are other things that help as well like hwaccel and just a well optimized setup.

**Reply by anthonyma94** [reply]
Thanks for the response. I haven't bought the camera yet, I just wanna get all the facts before I do. My server is running a Ryzen 7 5800X, so not lacking in the CPU department.

**Reply by NickM-27** [reply]
I see, unfortunately without a GPU or iGPU (I run ryzen 5700G) you won't be able to setup hwaccel for decoding the streams. You'll likely still be in good shape (again, depending on your expectations)

**Reply by anthonyma94** [reply]
Got it, thank you for your help!
